doc/source/whatsnew/v0.19.0.txt

```diff
@@ -457,6 +457,7 @@ API changes
 - ``pd.Timedelta(None)`` is now accepted and will return ``NaT``, mirroring ``pd.Timestamp`` (:issue:`13687`)
 - ``Timestamp``, ``Period``, ``DatetimeIndex``, ``PeriodIndex`` and ``.dt`` accessor have gained a ``.is_leap_year`` property to check whether the date belongs to a leap year. (:issue:`13727`)
 - ``pd.read_hdf`` will now raise a ``ValueError`` instead of ``KeyError``, if a mode other than ``r``, ``r+`` and ``a`` is supplied. (:issue:`13623`)
+- ``pd.read_csv()`` in the C engine will now issue a ``ParserWarning`` or raise a ``ValueError`` when ``sep`` encoded is more than one character long (:issue:`14065`)
```

Comment:
@jreback : This behaviour is consistent with whenever we passing a delimiter that has `len` > 1.  If you specifically say `engine='c'`, it will raise.  If you don't specify, because `engine` defaults to `c`, it will warn instead.  You can see that <a href="https://github.com/pydata/pandas/blob/8654a9ed3cc2246ef9eaf2fe8725369a2e885d35/pandas/io/parsers.py#L769">here</a> for other options that are unsupported by the C engine.